src/lib.rs

```diff
@@ -0,0 +1,253 @@
+// Copyright (c) 2020, Arm Limited, All Rights Reserved
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//          http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//! Client library for integration with the Parsec service
+#![deny(
+    nonstandard_style,
+    const_err,
+    dead_code,
+    improper_ctypes,
+    non_shorthand_field_patterns,
+    no_mangle_generic_items,
+    overflowing_literals,
+    path_statements,
+    patterns_in_fns_without_body,
+    private_in_public,
+    unconditional_recursion,
+    unused,
+    unused_allocation,
+    unused_comparisons,
+    unused_parens,
+    while_true,
+    missing_debug_implementations,
+    missing_docs,
+    trivial_casts,
+    trivial_numeric_casts,
+    unused_extern_crates,
+    unused_import_braces,
+    unused_qualifications,
+    unused_results,
+    missing_copy_implementations
+)]
+// This one is hard to avoid.
+#![allow(clippy::multiple_crate_versions)]
+
+pub mod auth;
+mod core;
+
+use crate::core::OperationHandler;
+use auth::AuthToken;
+use parsec_interface::operations::list_opcodes::Operation as ListOpcodes;
+use parsec_interface::operations::list_providers::{Operation as ListProviders, ProviderInfo};
+use parsec_interface::operations::ping::Operation as Ping;
+use parsec_interface::operations::psa_algorithm::AsymmetricSignature;
+use parsec_interface::operations::psa_destroy_key::Operation as PsaDestroyKey;
+use parsec_interface::operations::psa_export_public_key::Operation as PsaExportPublicKey;
+use parsec_interface::operations::psa_generate_key::Operation as PsaGenerateKey;
+use parsec_interface::operations::psa_import_key::Operation as PsaImportKey;
+use parsec_interface::operations::psa_key_attributes::KeyAttributes;
+use parsec_interface::operations::psa_sign_hash::Operation as PsaSignHash;
+use parsec_interface::operations::psa_verify_hash::Operation as PsaVerifyHash;
+use parsec_interface::operations::{NativeOperation, NativeResult};
+use parsec_interface::requests::Result;
+use parsec_interface::requests::{Opcode, ResponseStatus};
+use std::collections::HashSet;
+
+pub use parsec_interface::requests::ProviderID as Provider;
+
+/// Core client for Parsec service
+///
+/// The client exposes low-level functionality for using the Parsec service
+#[derive(Debug)]
+pub struct CoreClient {
```

Comment:
As the name of the structure suggests, I feel like it would belong better in the `core` module. Maybe we should leave `src/lib.rs` for re-exports, have `core` for exposing all our operations will all parameters so that it will be easier to expand with another module later which does more of smart-defaulting.

Reply:
I'd be ok with not implementing anything in `lib.rs` and just re-exporting from here